scripts/utils/termux_reuse_pr_build_artifacts.sh

```diff
@@ -0,0 +1,125 @@
+
+# fetch PR commit tree
+git fetch origin $PR_BASE_COMMIT:ref/tmp/$PR_BASE_COMMIT || error_exit "failed to fetch PR base tree, not performing CI fast path"
+git fetch origin $PR_HEAD_COMMIT:ref/tmp/$PR_HEAD_COMMIT || error_exit "failed to fetch PR head tree, not performing CI fast path"
+
+# make sure nobody force-pushed after CI was invoked and before we fetched remote tree and requested GraphQL data
+# checking if changed files in PR are same as in this push should be enough.
+
+diff -q \
+    <(git diff $PR_BASE_COMMIT $PR_HEAD_COMMIT | sed -n -E '/^diff --git a\// { p; n; /^index /!p; b } ; p') \
+    <(git diff $OLD_COMMIT $HEAD_COMMIT | sed -n -E '/^diff --git a\// { p; n; /^index /!p; b } ; p') >/dev/null 2>&1 \
+|| error_exit "PR head does not match pushed commit changes, probably PR ref was force pushed right after PR was merged. Not performing CI fast path."
+
+
+# obtain merge-base
+PR_MERGE_BASE="$(git merge-base ref/tmp/$PR_BASE_COMMIT $HEAD_COMMIT)" || error_exit "failed to obtain PR merge base, not performing CI fast path"
+
+# obtain list of all files changed in master branch since this PR diverged
+PR_BASE_TO_HEAD_CHANGED_FILES="$(git diff-tree --name-only -r $PR_MERGE_BASE..$OLD_COMMIT)" || true
+
+# obtain list of all packages changed by this PR
+PR_CHANGED_PACKAGES="$(
+  git diff-tree --name-only -r "$OLD_COMMIT..$HEAD_COMMIT" \
+    | grep -E "^($DIRS_REGEX)/[^/]+/" \
+    | sed -E "s#^(($DIRS_REGEX)/[^/]+)/.*#\1#" \
+    | sort -u
+)" || true
+
+# obtain list of all buildsystem files changed since this PR diverged
+PR_BASE_TO_HEAD_CHANGED_BUILDSYSTEM_FILES="$(
+  grep -e "^scripts/" -e "^ndk-patches/" -e "^build-package.sh$" <<< "$PR_BASE_TO_HEAD_CHANGED_FILES"
+)" || true
+
+# obtain list of all packages changes since this PR diverged
+PR_BASE_TO_HEAD_CHANGED_PACKAGES="$(
+  echo "$PR_BASE_TO_HEAD_CHANGED_FILES" \
+    | grep -E "^($DIRS_REGEX)/[^/]+/" \
+    | sed -E "s#^(($DIRS_REGEX)/[^/]+)/.*#\1#" \
+    | sort -u
+)" || true
+
+# obtain list of all build dependencies of packages changed by this PR
+PR_CHANGED_PACKAGES_DEPS="$(
+  for i in ${PR_CHANGED_PACKAGES:-}; do
+    ./scripts/buildorder.py $i $TERMUX_PACKAGE_DIRECTORIES | awk '{print $NF}'
+  done | sort -u
+)" || true
+
+# get list of all build dependencies changed since this PR diverged
+PR_BASE_TO_HEAD_CHANGED_DEPS="$(grep -Fxf <(echo "$PR_BASE_TO_HEAD_CHANGED_PACKAGES") - <<< "$PR_CHANGED_PACKAGES_DEPS")" || true
+
+echo "Packages changed by this PR:" ${PR_CHANGED_PACKAGES:-none}
+echo "Dependencies of these packages changed since PR divergence: " ${PR_BASE_TO_HEAD_CHANGED_DEPS:-}
+echo "Buildsystem files changed since PR divergence:" ${PR_BASE_TO_HEAD_CHANGED_BUILDSYSTEM_FILES:-none}
+echo "Packages updated since PR divergence:" ${PR_BASE_TO_HEAD_CHANGED_PACKAGES:-none}
+
+if [[ -z "${PR_BASE_TO_HEAD_CHANGED_BUILDSYSTEM_FILES:-}" && -z "${PR_BASE_TO_HEAD_CHANGED_DEPS:-}" ]]; then
```

Comment:
Since both need to be null, this is equivalent.
```suggestion
if [[ -z "${PR_BASE_TO_HEAD_CHANGED_BUILDSYSTEM_FILES:-}${PR_BASE_TO_HEAD_CHANGED_DEPS:-}" ]]; then
```